### PR TITLE
#170947612 Approvals Table

### PR DESCRIPTION
**Pull request by Niyonsengaeric**
#### What does this PR do?
- allow a user(administrator) to see a list of requests that require his/her approval on the Approvals page.
#### Description of Task to be completed?
- create approvals table.
- test the endpoint.
#### How should this be manually tested?
- run `git clone https://github.com/Stackup-Rwanda/knights-bn-frontend.git` to clone the repo.
- run `cd knights-bn-backend` in your terminal to enter into the repository.
- run `git fetch ` command to download commits, files, and refs from a remote repository into your local repo.
- run `git ft-approvals-table-170947612` to switch to this branch.
- run `npm install` to install dependencies.
- run `npm run dev-start` to start the application.
  you will see the login page then login with your administrator email or password or log in using your social account `Google & Facebook` 
- use `http://localhost:8080/pending` to access your approvals table.
#### Any background context you want to provide?

N/A

#### What are the relevant pivotal tracker stories?

[#170947612](https://www.pivotaltracker.com/story/show/170947612)
#### Screenshots (if appropriate)

       approvals table

Screen Shot 2020-05-26 at 08 28 26

         non-administrators not allowed to access the page.

Screen Shot 2020-05-26 at 08 27 57

### CODE REVIEW

**Comment by T2Wil** [comment]
Nice work @Niyonsengaeric ,
Here are few feedback :
1. I don't think these first two columns should be displayed

image

2. When I `as a manager` approve/reject the request the page should refresh with new data.

**Comment by Kagorora** [comment]
@Niyonsengaeric , Good Job
Here are my feedback
1. Fix the code climate issues
2. The table has too may information, could you reduce the columns that are in the table
and only remain with the necessary ones
3. Actions, accept or reject should be inline(not on top of the other)

**Review by T2Wil — APPROVED** [review]
Nice work @Niyonsengaeric

**Comment by Kagorora** [comment]
@Niyonsengaeric , Could you use Bootstrap Modal or any other to ask for confirmation, this default alert does not LGTM.

Screen Shot 2020-05-27 at 09 48 06

**Review by Kagorora — APPROVED** [review]

File: src/redux/actions/requestActions.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import * as types from './actionTypes';
```
[the remark this answers is not in the document]
Comment (kagabof) [reply src/redux/actions/requestActions.js]: Please check this error.

File: src/redux/reducers/requestReducer.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import * as types from '../actions/actionTypes';
```
[the remark this answers is not in the document]
Comment (kagabof) [reply src/redux/reducers/requestReducer.js]: Please check this error.

File: src/components/TableData.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
+import React from 'react';
```
[the remark this answers is not in the document]
Comment (kagabof) [reply src/components/TableData.js:3]: Please check this error.

File: src/components/TableData.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  const {
+    id, requesterId, type, reason, origin, destination, status, departureDate, returnDate, 
+    cities, createdAt, Comments,
+  } = request;
+  return (
+    <tr key={id}> <td>{id}</td> <td>{requesterId}</td> <td>{type}</td> <td>{reason}</td> <td>{origin}</td>
+    <td>{destination}</td> <td>{status}</td> <td>{convertDate(departureDate)}</td> <td>{convertDate(returnDate)}</td>
+    <td>{Array.isArray(cities) ? cities.map(({ name }) => `${name} - `) : `${origin} - ${destination}`} </td>
+      <td>{convertDate(createdAt)}</td> <td>{Comments}</td>
+      <td><a href="#" className="text-success" onClick={() => {
```
[the remark this answers is not in the document]
Comment (kagabof) [reply src/components/TableData.js:25]: Please check this error.

File: src/components/TableData.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+      if ( window.confirm('Are you sure you want to approve this request?'));}}>Approve</a>{' '}
+      <a href="#" className="text-danger" onClick={() => {
```
[the remark this answers is not in the document]
Comment (kagabof) [reply src/components/TableData.js:27]: Please check this error.

File: src/util/userRole.js [hunk 6]
```diff
@@ -0,0 +1,8 @@
+import jwt from 'jsonwebtoken';
```
[the remark this answers is not in the document]
Comment (kagabof) [reply src/util/userRole.js:1]: Please check this error.

File: src/views/PendingRequests.js [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+import { Table } from 'react-bootstrap';
+import { withRouter } from 'react-router';
+import '../assets/styles/approvals.scss';
+import { callApiThunk as thunk } from '../redux/thunk/index';
+import TableData from '../components/TableData';
+import { pendingRequests } from '../redux/actions/index';
+import userRole from '../util/userRole';
+import Alert from '../components/Alert';
+
+export const PendingRequest = (props) => {
```
[the remark this answers is not in the document]
Comment (kagabof) [reply src/views/PendingRequests.js:12]: Please check this error.

File: src/redux/reducers/requestReducer.js [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+
+const reduce = (state, action) => {
```
[the remark this answers is not in the document]
Comment (kagabof) [reply src/redux/reducers/requestReducer.js:3]: Please check this error.

File: src/views/PendingRequests.js [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
```
[the remark this answers is not in the document]
Comment (kagabof) [reply src/views/PendingRequests.js:12]: Please check this error.

File: src/components/TableData.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (kagabof) [reply src/components/TableData.js:3]: Please check this error.

File: src/components/TableData.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (kagabof) [reply src/components/TableData.js:25]: Please check this error.

File: src/redux/actions/requestActions.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (kagabof) [reply src/redux/actions/requestActions.js]: Please check this error.

File: src/redux/reducers/requestReducer.js (hunk 8 above)
[the remark this answers is not in the document]
Comment (kagabof) [reply src/redux/reducers/requestReducer.js:3]: Please check this error.

File: src/util/userRole.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (kagabof) [reply src/util/userRole.js:1]: Please check this error.

**Review by kagabof — CHANGES_REQUESTED** [review]

File: src/components/TableData.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (Niyonsengaeric) [reply src/components/TableData.js:3]: this file is also outdated maybe there is an issue with codeclimate!

**Review by kagabof — APPROVED** [review]
Well done.

**Pull request merged**